### PR TITLE
Price component

### LINKED ISSUE #11: Price component

### PR DESCRIPTION
**Pull request by mAiNiNfEcTiOn**
# What does this PR do:
Adds the Price component to Travix UI Kit.

Fixes #11.

screen shot 2017-02-22 at 17 11 35

# Where should the reviewer start:
  - Diffs
  - Look at the styleguide

# Unit and/or functional tests:
Adds Unit tests for Price component, reaching 100% coverage

### CODE REVIEW

**Comment by asci** [comment]
can you please provide a picture how it is looks like in LSG?

File: components/price/price.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  const rootClass = 'ui-price';
+  const className = getClassNamesWithMods(rootClass, mods);
+  const [intValue, decValue] = value.toString().split('.');
+
+  let formattedIntValue = intValue;
+  if (thousandsSeparator && (formattedIntValue.length > 3)) {
+    formattedIntValue = formattedIntValue
+      .split('')
+      .reverse()
+      .reduce((ret, digit, idx) => ret.concat([((idx > 0) && ((idx % 3) === 0) ? thousandsSeparator : ''), digit]), [])
```
Comment (asci) [inline components/price/price.js]: what do you think about extract this function? for example, name it 'removeThousandsSeparator'. IMHO would be better for readability.

File: components/price/price.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      .reverse()
+      .join('');
+  }
+
+  const formattedDecValue = decValue.length > decimalsPrecision
+    ? decValue.substr(0, decimalsPrecision)
+    : `${decValue}${'0'.repeat(decimalsPrecision - decValue.length)}`;
+
+  const dataAttributes = Object.keys(dataAttrs).reduce((ret, key) => {
```
Comment (asci) [inline components/price/price.js]: what about moving it to the `_helpers.js` file?

Comment (mAiNiNfEcTiOn) [inline components/price/price.js]: Will do, please stand-by :)

File: components/price/price.js (hunk 1 above)
Comment (mAiNiNfEcTiOn) [inline components/price/price.js]: 👍  Will do so...

File: components/price/price.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+const { PropTypes } = React;
+
+/**
+ * Adds the thousands separator to a given value.
+ * If either the thousands separator is not set or the value doesn't have thousands,
+ * it will return the original value.
+ *
+ * @function addThousandsSeparator
+ * @param {String} value              Integer value to be formatted
```
Comment (asci) [inline components/price/price.js]: it is not Integer

**Comment by asci** [comment]
Do we have only one size for Price component?

**Comment by mAiNiNfEcTiOn** [comment]
@asci again, the breakpoints are not supposed to be in the `travix-ui-kit`.

1. Not everyone has the same breakpoints
1. Not everyone wants to do responsive web design
1. Would not be flexible enough....

I would rather leave that for the a specific theme where you'd do something like:

```scss
.ui-price {
  .ui-price__currency {
    @include breakpoint('lt-medium') {
      font-size: 15px;
    }
  }
}
```

Or, if possible, override the variables would be even better.

File: components/price/price.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  const rootClass = 'ui-price';
+  const className = getClassNamesWithMods(rootClass, mods);
+  const [intValue, decValue] = value.toString().split('.');
+
+  const decimalsMarkup = showDecimals ? (<div className={`${rootClass}__decimals`}>{
+    decimalsSeparator + ensureDecimalPrecision(decValue, decimalsPrecision)
+  }</div>) : null;
+  const underlineMarkup = underlined ? <div className={`${rootClass}__underline`}/> : null;
+
+  return (<div className={className} {...getDataAttributes(dataAttrs)}>
```
Comment (AlexDudar) [inline components/price/price.js]: I would put html to the next line (outside of parentheses), to increase code readability

File: components/price/price.scss [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  flex-flow: row nowrap;
+  flex: 0 0 auto;
+}
+
+.ui-price__currency {
+  display: flex;
+  font-size: $tx-price-font-size-currency;
+  flex: 0 0 auto;
+
+  &.ui-price__currency-left {
```
Comment (AlexDudar) [inline components/price/price.scss]: what is `left` in context of this class? is it modifier or element's value?

**Comment by asci** [comment]
@mAiNiNfEcTiOn I'm not talking about responsiveness, I'm talking about semantics

File: components/price/price.scss (hunk 5 above)
Comment (mAiNiNfEcTiOn) [inline components/price/price.scss]: Well it's both a modifier and an element's value... But I'd go for modifier...

Should be separated by `_`, is that?

**Comment by mAiNiNfEcTiOn** [comment]
@asci define _semantics_ and what you're expecting and/or proposing please.

File: components/price/price.js (hunk 4 above)
Comment (mAiNiNfEcTiOn) [inline components/price/price.js]: I'm going to move it to the next line, but inside of the parenthesis :)

**Comment by asci** [comment]
@mAiNiNfEcTiOn  I mean we have different sizes for a button (small, medium, large). Why don't we have it for a price component? Maybe we have different sizes in design?

**Pull request merged**